File: fortran/emerytura.f90
program emerytura

implicit none


character :: tab
real :: summ, interest, rawsum, percent
integer :: years, monthSave, months, limit, i, k, j 
tab = char(5)

summ = 500
years = 10
monthSave = 500
months = 12
rawsum = summ
percent = 0.0
k = 0
limit = months*years

call random(years)

open(11,file="new.dat",status="OLD")
read(11,*), interest
do i=1,limit,1
    if (k > 11) then
        read(11,*), interest
        k = 0
        !print *,"in",interest
    end if
    !print *, "in= ", interest
    summ = summ + monthSave
    rawsum = rawsum + monthsave
    percent = summ*(interest/months)
    summ = summ + percent 
    k = k + 1
end do
close(11)

call spam(tab, years, monthSave, interest, summ, rawsum, months)
 
end program



subroutine spam(tab, years, monthSave, interest, summ, rawsum, months)

    character, intent(out) :: tab
    real, intent(out) :: summ, interest, rawsum
    integer, intent(out) :: years, monthSave, months

    print*," lat oszczedzania:", tab, years
    print*,"miesiecznie odkladane:", tab, monthSave
    print*,"oprocentowanie:", interest*100, "%"
    print*,"Uzbierana suma:", summ
    print*,"Z oprocentowaniem 0:", rawsum
    print*,"Zysk:", summ-rawsum
    print*,"+1000zl do pensji przez:", (summ/1000)/months, " lat"

end subroutine

subroutine random(years)
    integer :: n, j, clock
    integer :: years
    integer, dimension(:), allocatable :: seed
    real :: tmp,tmp2
    call random_seed(size=n)
    allocate(seed(n))
    call system_clock(COUNT=clock)
    open(10, file="new.dat", status="REPLACE")
    tmp = 0.01
    seed = clock + (/ (i - 1, i = 1, n) /)
    call random_seed(PUT=seed)
    do j=1, years+1, 1

        call random_number(tmp2)
        tmp2 = tmp2/10
        write(10,*), tmp2
        tmp = tmp + 0.01
    end do
    close(10)

end subroutine 

File: root_lab/atom/EMERYTURA/emerytura.cc
#include <iostream>
//#include <windows.h>
using namespace std;

int main(){

  double sum = 10000;
  int years = 45;
  int monthSave = 600;
  double interest = 0.018;
  //-------------------------------
  int months = 12;
  double rawsum = sum;
  double percent = 0.0;


  for(int i=0; i<months*years; i++){

    sum += monthSave;
    rawsum += monthSave; 
    percent = sum*(interest/months);
    sum += percent;

  }
  
  cout <<"# lat oszczedzania:\t"<< years<< endl;
  cout <<"miesiecznie odkladane:\t"<< monthSave <<endl;
  cout <<"oprocentowanie:\t\t"<< interest*100<<"%"<<endl;
  cout <<"Uzbierana suma:\t\t"<< sum << endl;
  cout <<"Z oprocentowaniem 0:\t"<< rawsum << endl;
  cout <<"Zysk:\t\t\t"<< sum-rawsum << endl;
  cout <<"+1000zl do pensji przez:\t" <<(sum/1000)/months <<" lat"<< endl;

//  Sleep(100000);
  return 0;
}
